azurecontainerapps.ts

```diff
@@ -201,9 +201,6 @@ export class azurecontainerapps {
         // Get the resource group to deploy to if it was provided, or generate it from the Container App name
         this.resourceGroup = await this.getOrCreateResourceGroup(this.containerAppName, this.location);
 
-        // Determine if the Container App currently exists
-        this.containerAppExists = await this.appHelper.doesContainerAppExist(this.containerAppName, this.resourceGroup);
-
         // If the Container App doesn't exist, get/create the Container App Environment to use for the Container App
         if (!this.containerAppExists) {
```

Comment:
I'm a bit confused about removing the `this.containerAppExists` code line 204-206. Isn't it going to have an impact on this condition here?

Reply:
Yeah we should keep this line but only execute it if the value of `this.containerAppExists` hasn't been set previously (we may need to make the variable a nullable boolean so we can differentiate between the default `false` value and no value set yet).

Reply:
I moved this condition to [this line](https://github.com/Azure/container-apps-deploy-action/blob/snehapar/Remove-hardcoded-location/azurecontainerapps.ts#L245) to make sure are not making this call twice. 

So, it should not impact the rest of the flow if this line is removed from here.

Reply:
We still need this line because if the user provides a value for `location`, then the call to `getLocation()` returns before it hits the part where it sets `this.containerAppExists` -- we just need to check if we've previously set a value for this variable or not because we don't want to call `doesContainerAppExist` multiple times.

Reply:
@snehapar9 I don't think we can rely on a guarantee on the order in which `setupResources` and `getLocation` will be called, as even if it is true today this could easily change (and no one would expect these to have a consequnce on each other).
So in this case, we could reach the line ` if (!this.containerAppExists) {` here and this would have an impact on the flow.

Just to be sure I got it, why did we need it to change things here? Could we handle this case in getLocation and check for the value there if it isn't already set?

Reply:
Synced offline. Reverted this change.